### DISCUSSION TITLE
Make zustand compatible with React 19

### DISCUSSION THREAD

**Discussion by dgtllion** [pmndrs/zustand#2842] [Q&A]
I see that zustand is supporting only <React18.
Is it possible to add React 19 in package.json? Or there are some work needed before that?

**Comment by dbritto-dev** [comment]
@dgtllion should work with React 19 without any issues but keep us in the loop if something went wrong

**Comment by dgtllion** [comment]
Thank you for your fast response.
I understand that it should work... What i mean when i try to add update React to 19 then i get incompatible warnings and one of them is zustand.

```
└─┬ zustand 5.0.1
  └── ✕ unmet peer @types/react@npm:types-react@19.0.0-rc.1: found 19.0.0-rc.1

```

**Comment by ispasdani** [comment]
Well if you for example want to use Zustand with NextJS 15, you get something like this.

<img width="853" alt="Screenshot 2024-11-15 094738" src="https://github.com/user-attachments/assets/92bbab74-a1bb-4b5b-b92d-b78d1d4808e2">

**Reply by dai-shi** [reply]
Can you try `--force`?

**Reply by JuanPabloQB1990** [reply]
lo instala falta ver si funciona bien los estados gracias...

**Reply by Didi3333** [reply]
Hi, can you update dependency version of use-sync-external-store to use the new one embeded react 19 => version 1.4.0

**Reply by dai-shi** [reply]
https://unpkg.com/browse/zustand@5.0.2/package.json
<img width="585" alt="image" src="https://github.com/user-attachments/assets/25e08b48-933e-4f6f-a563-8eb210b4306e" />
It should work with 1.4.0.

**Reply by Didi3333** [reply]
Yes it seems to work but it's to remove the warning 


![image](https://github.com/user-attachments/assets/4a487704-8346-4b68-908e-81a81d5ccf48)

**Reply by dai-shi** [reply]
It looks like you are using old version of zustand.

**Reply by YoniC0de** [reply]
I got the same error (with the latest zustand version 5.0.2)
`npm ERR! peer react@"^16.8.0 || ^17.0.0 || ^18.0.0" from use-sync-external-store@1.2.2
npm ERR! node_modules/use-sync-external-store
npm ERR!   peerOptional use-sync-external-store@">=1.2.0" from zustand@5.0.2
npm ERR!   node_modules/zustand
npm ERR!     zustand@"^5.0.2" from the root project
`

**Reply by dai-shi** [reply]
try `npm i use-sync-external-store@latest`
